topics/persistence.md

```diff
@@ -33,14 +39,16 @@ AOF advantages
 
 * Using AOF Redis is much more durable: you can have different fsync policies: no fsync at all, fsync every second, fsync at every query. With the default policy of fsync every second write performances are still great (fsync is performed using a background thread and the main thread will try hard to perform writes when no fsync is in progress.) but you can only lose one second worth of writes.
 * The AOF log is an append only log, so there are no seeks, nor corruption problems if there is a power outage. Even if the log ends with an half-written command for some reason (disk full or other reasons) the redis-check-aof tool is able to fix it easily.
-* Redis is able to automatically rewrite the AOF in background when it gets too big. The rewrite is completely safe as while Redis continues appending to the old file, a completely new one is produced with the minimal set of operations needed to create the current data set, and once this second file is ready Redis switches the two and starts appending to the new one.
 * AOF contains a log of all the operations one after the other in an easy to understand and parse format. You can even easily export an AOF file. For instance even if you've accidentally flushed everything using the `FLUSHALL` command, as long as no rewrite of the log was performed in the meantime, you can still save your data set just by stopping the server, removing the latest command, and restarting Redis again.
 
 AOF disadvantages
 ---
 
 * AOF files are usually bigger than the equivalent RDB files for the same dataset.
 * AOF can be slower than RDB depending on the exact fsync policy. In general with fsync set to *every second* performance is still very high, and with fsync disabled it should be exactly as fast as RDB even under high load. Still RDB is able to provide more guarantees about the maximum latency even in the case of an huge write load.
+* Using a lot of memory if there are writes to the database during a rewrite (these are buffered in memory and written to the new AOF at the end).
```

Comment:
Isn't this removed with the new muti-part AOF mechanism? Now we write the new writes to the new incremental file, don't we?

Reply:
i asked to add these 3 disadvantages, but it should be clear that these are issues relevant only to versions before 7.
maybe make a sub-section for pre-7.0 disadvantages?

Reply:
Yes, i think i should add **Redis < 7.0** and  **Redis >= 7.0** to splite them?

Reply:
not sure if we should split it all..
there could be some generic disadvantages that apply to all.
and then some version specific disadvantages

Reply:
@oranagra I think we should delete this sentence: `In the past we experienced rare bugs in specific commands (for instance there was one involving blocking commands like BRPOPLPUSH) causing the AOF produced to not reproduce exactly the same dataset on reloading. These bugs are rare and we have tests in the test suite creating ran`，We should talk about the disadvantages of AOF, not its previous bugs, WDYT?

Reply:
yes, i agree it's outdated...
i also think we can afford to delete this odd thing:
```
We have never had a single report from users about an AOF corruption that was detected in the real world.
  ```

Reply:
agree